basho/riak_cs
# Integrate bucket serialization

These changes make moss reliant on stanchion for bucket creation and deletion. Accordingly stanchion must be running and accessible for these bucket operations to be successfully exercised.
- Add infrastructure need for using stanchion.
- Remove duplicate `get_admin_creds` function.
- Fix some minor issues related to stanchion bucket creation requests.
- Changes to better handle error documents reported from stanchion.
- Change error codes to strings so xml error documents will be properly generated.
### Testing Suggestions
1. Start moss, but do not specify any admin credentials in the _app.config_ file. Attempt to create and delete and bucket and verify that `503 Service Unavailable` is returned. There are different variations on this theme that can be tried. _e.g._ Comment out the admin credentials in _app.config_ completely, comment out only one, specify one as a valid value and the other as an empty string.
2. Stop moss and specify admin credentials in the _app.config_ file. Start moss, but do not start stanchion. Attempt to create and delete and bucket and verify that `503 Service Unavailable` is returned.
3. Start stanchion. Attempt to create and delete a bucket and verify that it is successful. 
4. Recreate the bucket and then try to create the same bucket with another user's credentials. Verify that a `409` error is returned with an error code of `BucketAlreadyExists`.
5. Try to delete a nonexistent bucket and verify that a `404` error is returned with an error code of `NoSuchBucket`.
